aws-api/src/main/java/com/amplifyframework/api/aws/auth/ApiRequestSignerFactory.java

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amplifyframework.api.aws.auth;
+
+import androidx.annotation.NonNull;
+
+import com.amplifyframework.api.ApiException;
+import com.amplifyframework.api.aws.ApiAuthProviders;
+import com.amplifyframework.api.aws.AppSyncGraphQLRequest;
+import com.amplifyframework.api.aws.AuthorizationType;
+import com.amplifyframework.api.aws.sigv4.AppSyncV4Signer;
+import com.amplifyframework.api.graphql.GraphQLRequest;
+import com.amplifyframework.core.Amplify;
+import com.amplifyframework.logging.Logger;
+
+import com.amazonaws.Request;
+
+import java.util.Objects;
+import java.util.function.Supplier;
+
+/**
+ * Factory class that creates instances of different implementations of {@link ApiRequestSigner}s.
+ */
+public final class ApiRequestSignerFactory {
+    private static final Logger LOG = Amplify.Logging.forNamespace("amplify:aws-api");
+    private static final ApiRequestSigner NO_OP_REQUEST_SIGNER = new ApiRequestSigner() {
+        @Override
+        void addAuthHeader(Request<?> request) {
+        }
+    };
+
+    private final ApiAuthProviders apiAuthProviders;
+    private final String region;
+    private final AuthorizationType defaultAuthorizationType;
+
+    /**
+     * Constructor that accepts the API auth providers to be used with their respective request signer.
+     * @param apiAuthProviders An instance with fully configured auth providers for use when signing requests.
+     * @param defaultAuthorizationType The authorization type to use as default.
+     * @param region The AWS region where the API is deployed.
+     */
+    public ApiRequestSignerFactory(@NonNull ApiAuthProviders apiAuthProviders,
+                                   @NonNull AuthorizationType defaultAuthorizationType,
+                                   @NonNull String region) {
+        Objects.requireNonNull(apiAuthProviders);
+        Objects.requireNonNull(defaultAuthorizationType);
+        Objects.requireNonNull(region);
+        this.apiAuthProviders = apiAuthProviders;
+        this.defaultAuthorizationType = defaultAuthorizationType;
+        this.region = region;
```

Comment:
For short lines, you can write this more compactly -- `Objects.requireNonNull` returns the non-null value:
```suggestion
        this.apiAuthProviders = Objects.requireNonNull(apiAuthProviders);
        this.defaultAuthorizationType = Objects.requireNonNull(defaultAuthorizationType);
        this.region = Objects.requireNonNull(region);
```